### PR TITLE
Ipfs dockerize

### LINKED ISSUE #218: Publish Lunaria to IPFS
Use the Dockerfile as a starting point
https://notes.ethereum.org/@GW1ZUbNKR5iRjjKYx6_dJQ/Bk8zsJ9xj#As-a-Developer

### PR DESCRIPTION
**Pull request by jubalm**
Resolve #218 

#### Notable Changes
- use `debian` as base image
- installed kubo by copying `ipfs/kubo` binary

#### Testing

- build Lunaria locally with `docker build -t lunaria .`
- create a container with the built docker image
> $ docker container run --rm lunaria
added bafkreickot2jtfysomtkxw3wy2zdysmxcj2roshp33oebmmb4gpomqwc7e export/css/index.css
added bafkreibi4iqgpwoyncp2252fn2ynvikcdjbper2xsu5jbz4nqkyxjbe2p4 export/img/0x07865c6e87b9f70255377e024ace6630c1eaa37f.svg
...
added bafybeiajyjn6zx7rht6xxv37difjvl4oahjkmsfvikhabkobuzkq2utofe export

- check that ipfs hash matched the generated hash (from ipfs_hash.txt) during build
> $ docker run --entrypoint bin/sh lunaria -c "cat ipfs_hash.txt"
bafybeiajyjn6zx7rht6xxv37difjvl4oahjkmsfvikhabkobuzkq2utofe

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,29 @@
+FROM node:20-alpine as builder
+
+# Install app dependencies
+COPY ./package.json /source/package.json
+COPY ./package-lock.json /source/package-lock.json
+WORKDIR /source
+RUN npm ci
+
+# Copy project files and run necessary symlinks npm needs
```
Comment (MicahZoltu) [inline Dockerfile:9]:
```suggestion
# Run the vendoring script
```

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+RUN npm run build
+
+# Create kubo container
+FROM ipfs/kubo:latest as ipfs
+
+# Copy generated app
+COPY --from=builder /source/app /export
+
+# this entrypoint file will execute `ipfs add` of the build output to the docker host's IPFS API endpoint, so we can easily extract the IPFS build out of the docker image
+RUN printf '#!/bin/sh\nipfs add --cid-version 1 -r /export' >> /container-init.d/start.sh
```
Comment (MicahZoltu) [inline Dockerfile:29]: This should match the Tornado hack.md.  We will probably need to install a couple CLI tools to make that work though, up on line ~24.

**Review by MicahZoltu — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 2 above)
Comment (MicahZoltu) [reply Dockerfile:29]: This still needs to be addressed.  Ping me on Discord if you aren't sure how to do this.

**Review by MicahZoltu — CHANGES_REQUESTED** [review]
Still need the fix in the Dockerfile, everything else looks fine though.

File: .gitignore [hunk 3]
```diff
@@ -1,3 +1,4 @@
 /app/js/
 /app/vendor/
 node_modules/
+vendor/index.js
```
[the remark this answers is not in the document]
Comment (jubalm) [reply .gitignore]: removed from gitignore as it's handy to indeed check if the file is correctly removed by the vendor script

**Review by MicahZoltu — APPROVED** [review]

**Pull request merged**